### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by vladvovk2**

### CODE REVIEW

File: app/api/delivery_api/controllers/line_item_api.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+      resources :line_items do
+        before { authenticate! }
+
+        route_param :line_item_id, type: Integer do
+          desc 'Change product quantity.'
+          params { requires :quantity, type: Integer }
+          patch :quantity do
+            item = line_item.change_quantity(declared_params[:quantity])
+            present(message: "Quantity updated to #{params[:quantity]}") if item
```
Comment (slenderock) [inline app/api/delivery_api/controllers/line_item_api.rb]: Frontend always requires a machine-readable response which easies to process. 
So, I would suggest something like:
```
command = line_item.change_quantity(declared_params[:quantity])
present(message: "Quantity updated to #{declared_params[:quantity]}", quantity: declared_params[:quantity]) if command
```

File: app/api/delivery_api/controllers/order_api.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          end
+        end
+      end
+
+      resources :orders do
+        before { authenticate! }
+
+        desc 'Make order', entity: DeliveryApi::Entities::OrderResponce
+        params { use :order_params }
+        post :create do
```
Comment (slenderock) [inline app/api/delivery_api/controllers/order_api.rb]: no needs to specify `:create` here. According to REST, post to a plural name of resource means creating.

Let's change it to `post do`

File: config/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+default: &default
```
Comment (slenderock) [inline config/secrets.yml]: I guess we can change it to simple:
```
shared:
  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
  redis:
    url: <%= ENV["REDIS_URL"] %>
    namespace: <%= ENV["REDIS_NAMESPACE"] %>
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      - db
+      - redis
+    env_file:
+      - .env
+
+  redis:
+    container_name: kryla_api_redis
+    image: redis:4
+    ports:
+      - '6379:6379'
```
Comment (slenderock) [inline docker-compose.yml]: I would prefer to use double quotes in all .yml files.

File: lib/fog.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module Fog
```
Comment (slenderock) [inline lib/fog.rb]: WTF?

Comment (vladvovk2) [reply lib/fog.rb]: I have the following error when uploading to s3 "uninitialized constant CarrierWave::Storage::Fog" without this module. That trick resolve problem.

File: docker-compose.yml (hunk 4 above)
Comment (vladvovk2) [reply docker-compose.yml]: Switched to double quotes.

File: app/api/delivery_api/controllers/order_api.rb (hunk 2 above)
Comment (vladvovk2) [reply app/api/delivery_api/controllers/order_api.rb]: Yep, fixed :)

File: app/api/delivery_api/controllers/cart_api.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+module DeliveryApi
+  module Controllers
+    class CartApi < Root
+      helpers DeliveryApi::Helpers::CartHelpers
+
+      resources :carts do
```
[the remark this answers is not in the document]
Comment (vladvovk2) [reply app/api/delivery_api/controllers/cart_api.rb]: I'm fully agree with you about changes:
- resources :carts to resources :cart
- get :current_cart do to get do
But what about change post 'add_product/:product_id' do to put do. I agree with you that I need to remove  'add_product/:product_id'. I`am not sure about change method post to put cos:
 if cart.include? product
   quantity += 1   # => Put
else
   create             #  => Post
 end
Later i am rebuild this endpoint. maybe...

File: app/api/delivery_api/controllers/line_item_api.rb (hunk 1 above)
Comment (vladvovk2) [reply app/api/delivery_api/controllers/line_item_api.rb]: > Frontend always requires a machine-readable response which easies to process.
> So, I would suggest something like:
> 
> ```
> command = line_item.change_quantity(declared_params[:quantity])
> present(message: "Quantity updated to #{declared_params[:quantity]}", quantity: declared_params[:quantity]) if command
> ```

I like it. Fixed

File: app/api/delivery_api/controllers/order_api.rb (hunk 2 above)
Comment (vladvovk2) [reply app/api/delivery_api/controllers/order_api.rb]: > no needs to specify `:create` here. According to REST, post to a plural name of resource means creating.
> 
> Let's change it to `post do`

Yep, fixed :)

File: config/secrets.yml (hunk 3 above)
Comment (vladvovk2) [reply config/secrets.yml]: I have nothing against it

Comment (vladvovk2) [reply config/secrets.yml]: Yeah why not :)

**Pull request closed without merging**